brave/brave-core
# Allow to send from FEVM to typed FVM address

Allow to send from FEVM to typed FVM address
https://github.com/brave/brave-browser/issues/30490

To send form FEVM to FVM address FilForward contract with "forward" method is used.
Forward method takes byte form of Filecoin account (1 byte for protocol + payload).
To show right recipient when FF smart contract is used the same approach as for ERC20 is used,
only thing is that finalRecipient is added to TransactionInfo field for convenience.



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1) Create FEVM account
2) Create Filecoin account
3) Copy Filecoin account address to clipboard
4) Open send screen and select FEVM account, paste Filecoin account to the recipient field
5) Press send, ensure transaction completes
6) Test FEVM testnet -> FVM testnet
